src/server/server.js

```diff
@@ -33,25 +33,37 @@ expressApp.use(function (req, res, next) {
     if (result && result.failed) {
       next();
     } else {
-      var appConfig = {
-        context: context.getComponentContext()
-      };
-      var renderedApp = React.renderToString(fluxApp.getAppComponent()(appConfig));
-      var appState = 'window.appState = ' + serializeJavascript(fluxApp.dehydrate(context)) + ';';
-
-      var docConfig = {
-        title: 'Title',
-        markup: renderedApp,
-        state: 'window.state = ' + appState + ';'
-      };
-      var renderedHtml = '<!DOCTYPE html>' + 
-        React.renderToStaticMarkup(basePage(docConfig));
-
-      res.send(renderedHtml);
+      var route = context.getActionContext().router.getRoute(req.url);
+      if (route.config.actionPath) {
+        var pageActions = require('./actions/' + route.config.actionPath);
+        context.executeAction(pageActions.loadPage, { route: route }, function () {
+          renderApp(context, req, res);
+        });
+      } else {
+        renderApp(context, req, res);
+      }
     }
   });
 });
 
+function renderApp(context, req, res) {
+  var appConfig = {
+    context: context.getComponentContext()
+  };
+  var renderedApp = React.renderToString(fluxApp.getAppComponent()(appConfig));
+  var appState = 'window.appState = ' + serializeJavascript(fluxApp.dehydrate(context)) + ';';
+
+  var docConfig = {
+    title: 'Title',
+    markup: renderedApp,
+    state: 'window.state = ' + appState + ';'
```

Comment:
Does this mean you're doing `window.state = window.appState = ...;;`? Why do you need both?